pkg/skaffold/deploy/kpt/kpt.go

```diff
@@ -178,38 +177,79 @@ func (k *Deployer) Render(ctx context.Context, out io.Writer, builds []build.Art
 // adding image digests, and adding run-id labels.
 func (k *Deployer) renderManifests(ctx context.Context, _ io.Writer, builds []build.Artifact,
 	flags []string) (manifest.ManifestList, error) {
+	var err error
 	debugHelpersRegistry, err := config.GetDebugHelpersRegistry(k.globalConfig)
 	if err != nil {
 		return nil, fmt.Errorf("retrieving debug helpers registry: %w", err)
 	}
-	if k.Fn.SinkDir == "" {
-		k.Fn.SinkDir = tmpSinkDir
-	}
-
-	if err := os.RemoveAll(filepath.Join(k.Fn.SinkDir, k.Dir)); err != nil {
-		return nil, fmt.Errorf("deleting temporary directory %s: %w", filepath.Join(k.Fn.SinkDir, k.Dir), err)
-	}
-	if err := os.MkdirAll(filepath.Join(k.Fn.SinkDir, k.Dir), os.ModePerm); err != nil {
-		return nil, fmt.Errorf("creating temporary directory %s: %w", filepath.Join(k.Fn.SinkDir, k.Dir), err)
-	}
 
-	if err := k.readConfigs(ctx); err != nil {
+	var buf []byte
+	// Read the manifests under k.Dir as "source".
+	cmd := exec.CommandContext(
+		ctx, "kpt", kptCommandArgs(k.Dir, []string{"fn", "source"},
+			nil, nil)...)
+	buf, err = util.RunCmdOut(cmd)
+	if err != nil {
 		return nil, fmt.Errorf("reading config manifests: %w", err)
 	}
 
-	if err := k.kustomizeBuild(ctx); err != nil {
-		return nil, fmt.Errorf("kustomize build: %w", err)
+	// Hydrate the manifests source.
+	_, err = kustomize.FindKustomizationConfig(k.Dir)
+	// Only run kustomize if kustomization.yaml is found.
+	if err == nil {
+		// Note: A tmp dir is used to provide kustomize the manifest directory.
+		// Once the unified kpt/kustomize is done, kustomize can be run as a kpt fn step and
+		// this additional directory creation/deletion will no longer be needed.
+
+		err := os.MkdirAll(tmpKustomizeDir, os.ModePerm)
```

Comment:
is it ok to have an  existing `.kustomize` with some files in it?
Shd the code also check if there is a `.kustomize` and clean it up?

Reply:
sure.